io/plugins/eu.esdihumboldt.hale.io.csv.test/src/eu/esdihumboldt/hale/io/csv/CSVInstanceReaderTest.java

```diff
@@ -246,11 +280,47 @@ private Schema readCSVSchema(String sourceLocation, String typeName, String para
 		return schemaReader.getSchema();
 	}
 
+	private InstanceCollection readCSVInstances(String sourceLocation, String typeName, int skipN,
+			Schema sourceSchema, String seperator, String quote, String escape) throws Exception {
+
+		InstanceReader instanceReader = new CSVInstanceReader();
+		instanceReader.setSource(
+				new DefaultInputSupplier(getClass().getResource(sourceLocation).toURI()));
+		instanceReader.setParameter(CommonSchemaConstants.PARAM_TYPENAME, Value.of(typeName));
+		instanceReader.setParameter(CommonSchemaConstants.PARAM_SKIP_N_LINES, Value.of(skipN));
+		instanceReader.setParameter(CSVSchemaReader.PARAM_SEPARATOR, Value.of(seperator));
+		instanceReader.setParameter(CSVSchemaReader.PARAM_QUOTE, Value.of(quote));
+		instanceReader.setParameter(CSVSchemaReader.PARAM_ESCAPE, Value.of(escape));
+
+		instanceReader.setSourceSchema(sourceSchema);
+
+		// Test instances
+		IOReport report = instanceReader.execute(null);
+		assertTrue("Data import was not successfull.", report.isSuccess());
+
+		return instanceReader.getInstances();
+	}
+
 	private InstanceCollection readCSVInstances(String sourceLocation, String typeName,
 			boolean skipFirst, Schema sourceSchema, String seperator, String quote, String escape)
-					throws Exception {
-		return readCSVInstances(sourceLocation, typeName, skipFirst, sourceSchema, seperator, quote,
-				escape, null);
+			throws Exception {
+
+		InstanceReader instanceReader = new CSVInstanceReader();
```

Comment:
I believe that there were no additional changes in this method and the entire method body can be replaced by the original code from the master to avoid code duplication.

```
return readCSVInstances(sourceLocation, typeName, skipFirst, sourceSchema, seperator, quote,
 				escape, null);
```

Reply:
@kapil-agnihotri I don't fully understand your point. If I look at the code as it is now, it makes sense to me that all overloaded methods `readCSVInstances `return an object that is yet an instance, so I am fine with `return instanceReader.getInstances()`. What would it mean if instead I use `return readCSVInstances(sourceLocation, typeName, skipFirst, sourceSchema, seperator, quote, escape, null);`? Can you please explain it to me?

Reply:
> What would it mean if instead I use `return readCSVInstances(sourceLocation, typeName, skipFirst, sourceSchema, seperator, quote, escape, null);`


It will
- call another overloaded method with a `null` parameter for the last argument which will again return `instanceReader.getInstances()`
- avoid code duplication
- code logic in one place easier to modify and maintain, in case of any changes in the future, then the logic needs to be modified only in one place.